src/Polysemy/Internal/TH/Effect.hs

```diff
@@ -229,8 +245,11 @@ mkCLInfo dti ci = do
       cliEffArgs            = normalizeType <$> raw_cli_eff_args
       cliResType            = normalizeType     raw_cli_res_arg
       cliConName            = constructorName ci
-      cliFunName            = mkName $ mapHead toLower $ nameBase
-                            $ constructorName ci
+      cliFunName            = mkName
+                            $ (\case ':':cs -> cs
```

Comment:
Maybe this can be extracted in a named private function?

Reply:
Well, maybe it's not a bad idea - it seemed like something small enough to just be inlined, but now when I look at it...